src/plugins/intel_cpu/src/nodes/gather.cpp

```diff
@@ -137,27 +146,63 @@ void Gather::initSupportedPrimitiveDescriptors() {
         specIndicesSize = std::accumulate(idxDims.begin() + batchDims, idxDims.end(), 1lu, std::multiplies<Dim>());
 
         if (isDataShapeStat) {
+            specIdxAndAfterAxSize = specIndicesSize * afterAxisSize;
             specIdxAndAfterAxSizeB = specIndicesSize * afterAxisSizeInBytes;
             totalWork = beforeBatchSize * betweenBatchAndAxisSize * specIndicesSize * afterAxisSize;
         }
     }
 
-    // Implementation desc type will be redefined in the fn prepareParams if a kernel will be created.
     ov::element::Type dataPrecision = getOriginalInputPrecisionAtPort(GATHER_DATA);
+    if (weightsCompressed) {
+        if (!one_of(dataPrecision, ov::element::u8, ov::element::u4, ov::element::i8, ov::element::i4)) {
+            THROW_ERROR("has unsupported 'data' input precision: ", dataPrecision);
```

Comment:
We shouldn't throw exception in case of precision is out of supported list, instead we should set dataPrecision equal fp32 (as default fallback). CPU graph will automatically insert reorder to fp32 before GatherCompressed node in the graph.
Same true for other precision checks below.